extra/onnx_ops.py

```diff
@@ -89,6 +89,13 @@ def MaxPool(X, kernel_shape, auto_pad="NOTSET", ceil_mode=0, dilations=1, pads=N
   return _padding(X, pads, auto_pad, constant_value=-np.inf, axes=tuple(range(len(X.shape)))[-2:]).max_pool2d(kernel_shape, stride=strides, dilation=dilations)
 
 def Conv(X, W, B=None, auto_pad="NOTSET", dilations=1, group=1, kernel_shape=None, pads=None, strides=1):
+  if len(X.shape) == 3 and len(W.shape) == 3: # detect conv_1d conv
```

Comment:
Why doesn't this just work? I get the padding is wrong, but I expect conv2d to just work as a 1d conv the same way it just works as a 3d conv.

Reply:
Can you add 1d conv tests to test_ops without onnx?

Reply:
> Why doesn't this just work? I get the padding is wrong, but I expect conv2d to just work as a 1d conv the same way it just works as a 3d conv.

Well, I just assumed you need a (BxCxWxH) tensor for a conv2D and it's simple to turn a B x C x W -> B x C x W x 1. Does the conv2d actually work for conv3d tho? Wouldn't you have reshape your B x C x D x H x W tensor in some complicated way.

As for making it just work, without adding the extra dimension and then taking it away, I will look into it more. But from the first glance, it looks as the way conv2d is implemented, the output shape breaks when there is padding involved and contains negative dimensions.

> Can you add 1d conv tests to test_ops without onnx?

I will look into it.

Reply:
conv2d just works for conv2d, yea. See the test_ops.